bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/profile/ScriptProfile.java

```diff
@@ -92,18 +105,25 @@ public ProfileTypeUID getProfileTypeUID() {
 
     @Override
     public void onStateUpdateFromItem(State state) {
+        if (isConfigured && !toHandlerStateScript.isBlank()) {
+            internalFromItem(toHandlerStateScript, state);
+        }
     }
 
     @Override
     public void onCommandFromItem(Command command) {
-        if (isConfigured) {
-            String returnValue = executeScript(toHandlerScript, command);
-            if (returnValue != null) {
-                // try to parse the value
-                Command newCommand = TypeParser.parseCommand(handlerAcceptedCommandTypes, returnValue);
-                if (newCommand != null) {
-                    callback.handleCommand(newCommand);
-                }
+        if (isConfigured && !toHandlerCommandScript.isBlank()) {
+            internalFromItem(toHandlerCommandScript, command);
+        }
+    }
+
+    private void internalFromItem(String script, Type type) {
+        String returnValue = executeScript(script, type);
+        if (returnValue != null) {
+            // try to parse the value
+            Command newCommand = TypeParser.parseCommand(handlerAcceptedCommandTypes, returnValue);
+            if (newCommand != null) {
+                callback.handleCommand(newCommand);
```

Comment:
If we try to keep it as "open as possible", do we need to apply the same workaround here that we have in https://github.com/openhab/openhab-addons/pull/16263?

Reply:
Probably yes, but I am not sure whether we want to "distribute" that work-around across the code or if it would be better to wait for a solution of the underlying issue/problem.

Reply:
While having a look at `ProfileCallbackImpl`, the problem seems to be that `CommunicationManager::toAcceptedCommand` does not accept `OpenClosedType`, which IMO is correct as you normally do not command contact Items.

The question is, how to solve that.
I would think the cleanest solution is to add a method like `CommunicationManager::toAcceptedState` and to introduce a method called `handleState` or `handleUpdate` to the `ProfileCallback`. 
Then all profiles could call `handleUpdate` instead of `handleCommand` for state updates from the Item to the Thing.